src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,193 @@
-import React from 'react';
-
-export const TodoList: React.FC = () => (
-  <table className="table is-narrow is-fullwidth">
-    <thead>
-      <tr>
-        <th>#</th>
-        <th>
-          <span className="icon">
-            <i className="fas fa-check" />
+import React, { useState } from 'react';
+
+import classNames from 'classnames';
+
+import { TodoModal } from '../TodoModal';
+
+import { Todo } from '../../types/Todo';
+
+enum SortType {
+  All = 'all',
+  Completed = 'completed',
+  Active = 'active',
+}
+
+function getReorderedTodos(
+  todos: Todo[],
+  sortType: SortType,
+): Todo[] {
+  const visibleTodos = [...todos];
+
+  switch (sortType) {
+    case SortType.Active:
+      return visibleTodos.filter(todo => {
+        return todo.completed === false;
+      });
+
+    case SortType.Completed:
+      return visibleTodos.filter(todo => {
+        return todo.completed === true;
+      });
+
+    default:
+      return visibleTodos;
+  }
+}
+
+type Props = {
+  todos: Todo[];
+};
+
+export const TodoList: React.FC<Props> = ({ todos }) => {
+  const [selectedId, setSelectedId] = useState(0);
+  const [query, setQuery] = useState('');
+  const [sortType, setSortType] = useState(SortType.All);
+
+  const visibleTodos = getReorderedTodos(
+    todos,
+    sortType,
+  );
+
+  const filteredTodos = visibleTodos.filter(todo => {
+    const fixedQuery = query.toLowerCase();
+
+    return todo.title.includes(fixedQuery);
+  });
+
+  return (
+    <>
+      <form className="field has-addons">
+        <p className="control">
+          <span className="select">
+            <select
+              data-cy="statusSelect"
+              value={sortType}
+              onChange={({ target }) => {
+                // eslint-disable-next-line no-nested-ternary
+                return target.value === 'all'
+                  ? setSortType(SortType.All)
+                  : target.value === 'active'
+                    ? setSortType(SortType.Active)
+                    : setSortType(SortType.Completed);
+              }}
+            >
+              <option
+                value="all"
+              >
+                All
+              </option>
+              <option
+                value="active"
+              >
+                Active
+              </option>
+              <option
+                value="completed"
+              >
+                Completed
+              </option>
+            </select>
+          </span>
+        </p>
+
+        <p className="control is-expanded has-icons-left has-icons-right">
+          <input
+            data-cy="searchInput"
+            type="text"
+            className="input"
+            placeholder="Search..."
+            value={query}
+            onChange={(event) => {
+              return setQuery(event.target.value);
+            }}
```

Comment:
setQuery: (value) => void

return setQuery() // returns void

```suggestion
            onChange={(event) => {
              setQuery(event.target.value);
            }}
```
also returns void

last one is correct)